staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go

```diff
@@ -49,95 +49,101 @@ func NestedFieldCopy(obj map[string]interface{}, fields ...string) (interface{},
 // to traverse obj.
 //
 // Note: fields passed to this function are treated as keys within the passed
-// object; no array/slice syntax is supported.
-func NestedFieldNoCopy(obj map[string]interface{}, fields ...string) (interface{}, bool, error) {
-	var val interface{} = obj
+// object.
+func NestedFieldNoCopy(obj map[string]any, fields ...string) (any, bool, error) {
+	var val any = obj
 
 	for i, field := range fields {
 		if val == nil {
 			return nil, false, nil
 		}
-		if m, ok := val.(map[string]interface{}); ok {
+		if m, ok := val.(map[string]any); ok {
 			val, ok = m[field]
 			if !ok {
 				return nil, false, nil
 			}
 		} else {
-			return nil, false, fmt.Errorf("%v accessor error: %v is of the type %T, expected map[string]interface{}", jsonPath(fields[:i+1]), val, val)
+			return nil, false, fmt.Errorf("%v accessor error: %v is of the type %T, expected map[string]any", jsonPath(fields[:i+1]), val, val)
 		}
 	}
 	return val, true, nil
 }
 
-// NestedString returns the string value of a nested field.
-// Returns false if value is not found and an error if not a string.
-func NestedString(obj map[string]interface{}, fields ...string) (string, bool, error) {
-	val, found, err := NestedFieldNoCopy(obj, fields...)
+// NestedValueCopy returns a deep copy of the value of a nested field
+// which has desired type. Returns false if the value is missing.
```

Comment:
nit: preexisting from other functions too, but missing from this doc and the other doc is the expected return value of `found` if the value was present, but there was an error casting

Reply:
Comments have been made clearer.